### PR TITLE
GraphQL Helix and Envelop @redwoodjs/api default

### PR DESCRIPTION
**Pull request by dthyresson**
Fixes https://github.com/redwoodjs/redwood/issues/3271

RedwoodJS v0.37 aims to sunset the use of Apollo Server as the GraphQL server and instead use GraphQL Helix and Envelop that has been an experimental preview feature.

To this end, first need to repackage GraphQL Helix and Envelop into `api` from `graphql-server`:

- [x] Deprecate `@graphql-server` package in v0.37 and remove each of a release candidate (v.38?)
- [x] Migrate `@graphql-server` use of Helix and envelop to `api` package
- [x] Ensure e2e tests pass

Rationale for removing graphql-server:

* Clear packaging - your api is serverless functions, services, and your graphql endpoint
* Single package for where Global Context lives is critical because we have seen that the Global Context *can* get confused if set in Envelop-based AuthContext such that using context in services leads to authentication issues
* Ease transition for existing apps using api
* Fewer documentation changes. Do not need to explain api vs graphql-server and how, why and when to use
* Less dependencies (api doesn't depend of graphql-server for the graphql implemetation and graphql-server doesn't depend off api for logger, webhooks, etc
* Easier to manage and generate envelop plugins and GraphQL directives

There are other tasks related to this effort, but will be created as separate issues and PR's.

This includes 

- refactor use of `apollo-server-lambda` in errors.
- remove logic in dev server for `useEnvelop` experimental feature
- remove Helix/Envelop e2e integration scenario

Not done:

- [ ] How to test that `graphql-server` is really deprecated?

### CODE REVIEW

**Comment by dthyresson** [comment]
@dac09 and @thedavidprice can I get your review on this ahead of getting Peter's review.

File: packages/api/package.json [hunk 1]
```diff
@@ -10,27 +10,37 @@
...
+    "@envelop/disable-introspection": "1.0.0",
+    "@envelop/filter-operation-type": "1.0.0",
+    "@envelop/parser-cache": "1.0.0",
+    "@envelop/validation-cache": "1.0.0",
     "@graphql-tools/merge": "7.0.0",
+    "@graphql-tools/schema": "8.0.3",
+    "@graphql-tools/utils": "8.0.2",
     "@prisma/client": "2.29.1",
     "@types/pino": "6.3.11",
     "apollo-server-lambda": "2.25.2",
```
Comment (dthyresson) [inline packages/api/package.json:25]: Note that we need to import `apollo-server-lambda` temporarily due to the use of Apollo Server errors and mimicking their behaviors in the error handling plugins for the GraphQL handler.

The `apollo-server-lambda` import will be removed once error handling on the graphql and web-side formError handling is refactored.

File: packages/api/src/functions/__tests__/graphql.test.ts [hunk 2]
```diff
@@ -11,21 +27,19 @@ describe('global context handlers', () => {
     process.env.SAFE_GLOBAL_CONTEXT = '0'
   })
 
-  it('merges the apollo resolver and global context correctly', async () => {
+  it('merges the graphql-server resolver and global context correctly', async () => {
```
Comment (dthyresson) [inline packages/api/src/functions/__tests__/graphql.test.ts:30]: I should no longer refer to graphql-server or Apollo -- they are simply resolvers.

```suggestion
  it('merges the  resolver and global context correctly', async () => {
```

File: packages/api/src/functions/__tests__/graphql.test.ts [hunk 3]
```diff
@@ -65,21 +77,19 @@ describe('global context handlers', () => {
...
       c: 25,
       d: 5,
-      callbackWaitsForEmptyEventLoop: false,
     })
   })
 })
 
 describe('per request context handlers', () => {
-  it('merges the apollo resolver and global context correctly', async () => {
+  it('merges the graphql-server resolver and global context correctly', async () => {
```
Comment (dthyresson) [inline packages/api/src/functions/__tests__/graphql.test.ts:92]:
```suggestion
  it('merges the resolver and global context correctly', async () => {
```

File: packages/graphql-server/package.json [hunk 4]
```diff
@@ -10,40 +10,11 @@
...
-    "jwks-rsa": "2.0.4",
-    "lodash.merge": "4.6.2",
-    "lodash.omitby": "4.6.0",
-    "pino": "6.13.0",
-    "pino-pretty": "5.1.3",
-    "uuid": "8.3.2"
+    "@redwoodjs/api": "0.36.0"
   },
   "devDependencies": {
     "@envelop/types": "1.0.2",
```
Comment (dthyresson) [inline packages/graphql-server/package.json:16]:
```suggestion
```

Can remove `@envelop/types`, but eventually the entire package will be.

File: packages/graphql-server/src/index.ts [hunk 5]
```diff
@@ -1,11 +1,16 @@
-// In transition can keep Apollo Server error compatibility
-// export * from './errors'
+/**
+ * @deprecated
```
Comment (dthyresson) [inline packages/graphql-server/src/index.ts:2]: @dac09 I am not sure how best the make everything here deprecated -- not how to test that it is,

**Comment by dthyresson** [comment]
See questions about an alternate approach -- to deprecate api and to use graphql-server here:

https://github.com/redwoodjs/redwood/issues/3271#issuecomment-904235486

**Comment by peterp** [comment]
@dthyresson @dac09 @thedavidprice Please go ahead and merge this without my review - I don't think I can add any additional context here and don't want to slow this down.

**Comment by dthyresson** [comment]
> @dthyresson @dac09 @thedavidprice Please go ahead and merge this without my review - I don't think I can add any additional context here and don't want to slow this down.

No worries. 

I need to rework this actually to be in sync with what we discussed and confirmed on the issue -- namely, deprecate the graph portion of the api and keep graphql-server.

**Comment by dthyresson** [comment]
On second thought, I am going to close and make a new PR since it will be easier not to move back all the graphql-server code.

**Pull request closed without merging**